pkg/status/combinedstatus-resolution.go

```diff
@@ -397,9 +429,10 @@ func evaluateWorkStatusAgainstStatusCollectorWriteLocked(celEvaluator *celEvalua
 	// evaluate select
 	selectEvals := make(map[string]ref.Val)
 	for _, selectNamedExp := range scData.Select {
-		eval, err := celEvaluator.Evaluate(selectNamedExp.Def, workStatusContent)
+		eval, err := celEvaluator.Evaluate(selectNamedExp.Def, content)
 		if err != nil {
-			return false, fmt.Errorf("failed to evaluate select expression: %w", err)
+			runtime2.HandleError(fmt.Errorf("failed to evaluate select expression: %w", err))
+			continue
 		}
 
 		updated = updated || eval.Equal(wsData.selectEval[selectNamedExp.Name]).Value() != true
```

Comment:
`wsData.selectEval` could be `nil`, right? And `wsData.selectEval[selectNamedExp.Name]` could be `nil`, right?

Reply:
I looked at the definition of the `Equal` method of the `Val` interface, and did not notice a promise about how it behaves if `other Val` is `nil`. I think that it would be best if KubeStellar did not rely on undocumented behavior in this case.